With this patch, build fails with a different error
```
/build/krita-plugin-gmic/src/gmic-3.2.1.1-patched/gmic-qt/src/PersistentMemory.cpp: In static member function ‘static void GmicQt::PersistentMemory::clear()’:
/build/krita-plugin-gmic/src/gmic-3.2.1.1-patched/gmic-qt/src/PersistentMemory.cpp:43:17: error: no matching function for call to ‘gmic_library::gmic_image<char>::assign()’
   43 |   image().assign();
      |   ~~~~~~~~~~~~~~^~
In file included from /build/krita-plugin-gmic/src/gmic-3.2.1.1-patched/gmic-qt/src/PersistentMemory.cpp:26:
/usr/include/gmic.h:111:20: note: candidate: ‘template<class t> gmic_library::gmic_image<T>& gmic_library::gmic_image<T>::assign(const t*, unsigned int, unsigned int, unsigned int, unsigned int) [with T = char]’
  111 |     gmic_image<T>& assign(const t *const values, const unsigned int size_x, const unsigned int size_y=1,
      |                    ^~~~~~
/usr/include/gmic.h:111:20: note:   template argument deduction/substitution failed:
/build/krita-plugin-gmic/src/gmic-3.2.1.1-patched/gmic-qt/src/PersistentMemory.cpp:43:17: note:   candidate expects 5 arguments, 0 provided
   43 |   image().assign();
      |   ~~~~~~~~~~~~~~^~
/usr/include/gmic.h:106:20: note: candidate: ‘gmic_library::gmic_image<T>& gmic_library::gmic_image<T>::assign(unsigned int, unsigned int, unsigned int, unsigned int) [with T = char]’
  106 |     gmic_image<T>& assign(const unsigned int size_x, const unsigned int size_y=1,
      |                    ^~~~~~
/usr/include/gmic.h:106:20: note:   candidate expects 4 arguments, 0 provided
/usr/include/gmic.h:114:20: note: candidate: ‘gmic_library::gmic_image<T>& gmic_library::gmic_image<T>::assign(const T*, unsigned int, unsigned int, unsigned int, unsigned int, bool) [with T = char]’
  114 |     gmic_image<T>& assign(const T *const values, const unsigned int size_x, const unsigned int size_y,
      |                    ^~~~~~
/usr/include/gmic.h:114:20: note:   candidate expects 6 arguments, 0 provided
```

--- a/gmic-qt/CMakeLists.txt
+++ b/gmic-qt/CMakeLists.txt
@@ -562,12 +562,42 @@ set (gmic_qt_FORMS
 
 if(ENABLE_DYNAMIC_LINKING)
   set(CMAKE_SKIP_RPATH TRUE)
-  set(gmic_qt_LIBRARIES
-    ${gmic_qt_LIBRARIES}
-    "gmic"
+  # G'MIC-Qt needs visibility into the private symbols defined
+  # by the gmic.cpp plugin. However, this is only possible
+  # if the library is static OR if it's dynamic and built by
+  # a compiler that supports .so-style exports.
+  if (MSVC OR NOT ENABLE_SYSTEM_GMIC)
+    set(gmic_qt_LIBRARIES
+      ${gmic_qt_LIBRARIES}
+      gmicstatic
     )
-  if (NOT ENABLE_SYSTEM_GMIC)
-    link_directories(${GMIC_LIB_PATH})
+    # Mimic an external G'MIC library build for catching link ABI errors.
+    add_library(gmicstatic STATIC ../src/gmic.cpp)
+    target_include_directories(gmicstatic PUBLIC ../src)
+    # We need internal access into the gmic-core API.
+    target_compile_definitions(gmicstatic PUBLIC gmic_core)
+    set_target_properties(gmicstatic
+      PROPERTIES
+        AUTOMOC OFF
+    )
+    target_link_libraries(gmicstatic PUBLIC
+      ${PNG_LIBRARIES}
+      ${FFTW3_LIBRARIES}
+      ${ZLIB_LIBRARIES}
+      ${CURL_LIBRARIES}
+      ${EXTRA_LIBRARIES})
+  elseif(TARGET libgmicstatic OR TARGET libgmic OR GMIC_LIB_PATH)
+    set(gmic_qt_LIBRARIES
+      ${gmic_qt_LIBRARIES}
+      "gmic"
+    )
+    if (GMIC_LIB_PATH)
+      link_directories(${GMIC_LIB_PATH})
+    endif()
+    # Inject the G'MIC CImg plugin.
+    include_directories(../src)
+  else()
+    message(FATAL_ERROR "No G'MIC library is available for linking. Please build libgmic as a static library.")
   endif()
 else(ENABLE_DYNAMIC_LINKING)
   set(gmic_qt_SRCS

